File: backend/Repositories/Enums/UserPermissions.cs
namespace Repositories.Enums;

[Flags]
public enum UserPermissions
{
    None = 0b_0000_0000,
    SubmitDeck = 0b_0000_0001,
    SubmitCustomCard = 0b_0000_0010,
    RateDeck = 0b_0000_0100,
    RateCustomCard = 0b_0000_1000,
    ValidateCustomCard = 0b_0001_0000,
    SetDeckOfTheDay = 0b_0010_0000,
    SetCustomCardOfTheDay = 0b_0100_0000,
    ChangePermissions = 0b_1000_0000,
    Normal = SubmitDeck | SubmitCustomCard | RateDeck | RateCustomCard,
    Moderator = Normal | SetDeckOfTheDay | SetCustomCardOfTheDay | ValidateCustomCard,
    Admin = Moderator | ChangePermissions
}

File: frontend/src/enums/UserPermissions.ts
enum UserPermissions {
    None = 0,
    SubmitDeck = 1,
    SubmitCustomCard = 1 << 1,
    RateDeck = 1 << 2,
    RateCustomCard = 1 << 3,
    ValidateCustomCard = 1 << 4,
    SetDeckOfTheDay = 1 << 5,
    SetCustomCardOfTheDay = 1 << 6,
    ChangePermissions = 1 << 7,
    Normal = SubmitDeck | SubmitCustomCard | RateDeck | RateCustomCard,
    Moderator = Normal | SetDeckOfTheDay | SetCustomCardOfTheDay | ValidateCustomCard,
    Admin = Moderator | ChangePermissions
}

export { UserPermissions }